src/dlsproto/client/request/internal/GetRange.d

```diff
@@ -447,19 +478,19 @@ private scope class GetRangeHandler
 
         public void addRecords ( in void[] record_batch )
         {
+            auto batch_buffer = this.buffers.input;
+
             // append record_batch to *this.batch_buffer, which may or may not
             // be empty.
-
-            if (!(*this.batch_buffer).length)
-                enableStomping(*this.batch_buffer);
+            if (!(*batch_buffer).length)
```

Comment:
Would seem sensible (and simple) to encapsulate this in some kind of `push` method.